### PR TITLE
[WIP] Box build pipeline

### PR DESCRIPTION
**Pull request by yzl**
cc @fnichol @schisamo @tyler-ball 

Build a set of boxes based on the most recently changed files in git, upload the box artifacts to atlas, and release them.

- [x] Tested against templates for public boxen.
- [x] Support proprietary/private boxen.
- [ ] The Rakefile is dreadful because rake makes me feel like I have more than 8 thumbs.  Suggestions on how to make it better would be welcome.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -0,0 +1,239 @@
+require 'cgi'
+require 'json'
+require 'net/http'
+
+# Enables `bundle exec bundle exec rake do_all[ubuntu-12.04-amd64,centos-7.1-x86_64]
```
Comment (tyler-ball) [inline Rakefile]: (nit) `bundle exec` twice

**Comment by schisamo** [comment]
Just an FYI that first-class WinRM support is now part of Packer core:
https://twitter.com/mitchellh/status/610167963673473024
https://github.com/mitchellh/packer/issues/451

This should make the Windows base box builds just work.

**Comment by cheeseplus** [comment]
Given the additional work I've done to validate this approach I'm :+1: on merging this and then I'll apply my changes on top. This is more than good enough for an MVP and we can lament or toast our choice of Rake in hindsight.

**Comment by fnichol** [comment]
Woo :+1:

**Pull request closed without merging**